File: src/main/java/com/gmail/borlandlp/minigamesdtools/gun/bullet/GhostBulletCreator.java
package com.gmail.borlandlp.minigamesdtools.gun.bullet;

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.Creator;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo;
import org.bukkit.World;

import java.util.Arrays;
import java.util.List;

@CreatorInfo(creatorId = "ghost_bullet")
public class GhostBulletCreator implements Creator {
    @Override
    public List<String> getDataProviderRequiredFields() {
        return Arrays.asList("world");
    }

    @Override
    public Object create(String ID, AbstractDataProvider dataProvider) throws Exception {
        return new GhostBullet((World) dataProvider.get("world"));
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/gun/bullet/GhostBulletCreator.kt
package com.gmail.borlandlp.minigamesdtools.gun.bullet

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.Creator
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo
import org.bukkit.World

@CreatorInfo(creatorId = "ghost_bullet")
class GhostBulletCreator : Creator() {
    override val dataProviderRequiredFields: List<String>
        get() = listOf("world")

    @Throws(Exception::class)
    override fun create(id: String, dataProvider: AbstractDataProvider): Any {
        return GhostBullet(dataProvider["world"] as World)
    }
}